cmd/cluster-agent/api/server.go

```diff
@@ -46,7 +47,7 @@ var (
 )
 
 // StartServer creates the router and starts the HTTP server
-func StartServer(senderManager sender.DiagnoseSenderManager) error {
+func StartServer(senderManager sender.DiagnoseSenderManager, leaderForwarder *api.LeaderForwarder) error {
```

Comment:
We pass down the same leader forwarder to both the CLC handler and the PLD handler. But we could have a global one or separate ones. Any opinion here ?